src/readfish/plugins/mappy.py

```diff
@@ -43,6 +43,20 @@ def __init__(
         self.logger = setup_debug_logger(__name__, log_file=debug_log)
         self.aligner_params = kwargs
         self.aligner = mappy.Aligner(**self.aligner_params)  # type: ignore
+        if not self.initialised:
+            """We will test to see if the reference file exists"""
+            if "fn_idx_in" in self.aligner_params:
+                """We will test to see if the reference file exists"""
+                import os
```

Comment:
We prefer pathlib - we use in most of the rest of the codebase
Also import at the top of the file